### PR TITLE
Add widget refactor

### PR DESCRIPTION
**Pull request by Dave-Gallup**
This is linted but lacks testing due to the issue Meghan faced.  Tests will hopefully be added later for this portion of the code.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -6,6 +6,7 @@
     "@databraid/github-widget": "1.0.5",
     "@databraid/slack-widget": "1.0.3",
     "@databraid/transit-widget": "1.0.16",
+    "prop-types": "^15.5.10",
```
Comment (HamidAghdaee) [inline package.json]: please exclude ^

File: src/App.jsx [hunk 2]
```diff
@@ -1,43 +1,138 @@
...
-      h: 10,
-      minH: 6.5,
-      minW: 7,
-    },
-    { i: 'b', x: 3, y: 0, w: 6, h: 12 },
-    { i: 'c', x: 0, y: 0, w: 6, h: 10 },
-  ];
+const App = (props) => {
+  const components = (props.ids).map((component) => {
+    if (component === TRANSIT_WIDGET_ID) {
```
Comment (HamidAghdaee) [inline src/App.jsx]: let's rename component to widgetId

File: package.json (hunk 1 above)
Comment (Dave-Gallup) [reply package.json]: done

File: src/App.jsx (hunk 2 above)
Comment (Dave-Gallup) [reply src/App.jsx]: changed 'component' to 'widgetId'

**Pull request merged**